src/main/java/frc/robot/subsystems/HoodSubsystem.java

```diff
@@ -0,0 +1,169 @@
+    m_motor.setIdleMode(IdleMode.kBrake);
+    Controller = new XboxController(0);
+
+    /**
+     * The restoreFactoryDefaults method can be used to reset the configuration
+     * parameters
+     * in the SPARK MAX to their factory default state. If no argument is passed,
+     * these
+     * parameters will not persist between power cycles
+     */
+    m_motor.restoreFactoryDefaults();
+
+    /**
+     * In order to use PID functionality for a controller, a SparkMaxPIDController
+     * object
+     * is constructed by calling the getPIDController() method on an existing
+     * CANSparkMax object
+     */
+    m_pidController = m_motor.getPIDController();
+
+    // Encoder object created to display position values
+    m_encoder = m_motor.getEncoder();
+
+    // PID coefficients
+    kP = 0.001;
+    kI = 0.00001;
+    kD = 0;
+    kIz = 0;
+    kFF = 0;
+    kMaxOutput = 1;
+    kMinOutput = -1;
+    
+    
+
+    // set PID coefficients
+    m_pidController.setP(kP);
+    m_pidController.setI(kI);
+    m_pidController.setD(kD);
+    m_pidController.setIZone(kIz);
+    m_pidController.setFF(kFF);
+    m_pidController.setOutputRange(kMinOutput, kMaxOutput);
+
+
+    // display PID coefficients on SmartDashboard
+    SmartDashboard.putNumber("P Gain", kP);
+    SmartDashboard.putNumber("I Gain", kI);
+    SmartDashboard.putNumber("D Gain", kD);
+    SmartDashboard.putNumber("I Zone", kIz);
+    SmartDashboard.putNumber("Feed Forward", kFF);
+    SmartDashboard.putNumber("Max Output", kMaxOutput);
+    SmartDashboard.putNumber("Min Output", kMinOutput);
+    SmartDashboard.putNumber("Set Rotations", rotations);
+
+  }
+
+  @Override
+  public void periodic() {
+    // This method will be called once per scheduler run
+    // read PID coefficients from SmartDashboard
+    double p = SmartDashboard.getNumber("P Gain", 0);
+    double i = SmartDashboard.getNumber("I Gain", 0);
+    double d = SmartDashboard.getNumber("D Gain", 0);
+    double iz = SmartDashboard.getNumber("I Zone", 0);
+    double ff = SmartDashboard.getNumber("Feed Forward", 0);
+    double max = SmartDashboard.getNumber("Max Output", 0);
+    double min = SmartDashboard.getNumber("Min Output", 0);
+    //double rotations = SmartDashboard.getNumber("Set Rotations", 0);
+    
+ 
+    // if PID coefficients on SmartDashboard have changed, write new values to
+    // controller
+    if ((p != kP)) {
+      m_pidController.setP(p);
+      kP = p;
+    }
+    if ((i != kI)) {
+      m_pidController.setI(i);
+      kI = i;
+    }
+    if ((d != kD)) {
+      m_pidController.setD(d);
+      kD = d;
+    }
+    if ((iz != kIz)) {
+      m_pidController.setIZone(iz);
+      kIz = iz;
+    }
+    if ((ff != kFF)) {
+      m_pidController.setFF(ff);
+      kFF = ff;
+    }
+    if ((max != kMaxOutput) || (min != kMinOutput)) {
+      m_pidController.setOutputRange(min, max);
+      kMinOutput = min;
+      kMaxOutput = max;
+    }
+      
+    if(Controller.getAButtonPressed()){
```

Comment:
This is great for testing but eventually we will want this to be a method to be called by a command